Framework/Algorithms/src/ModeratorTzero.cpp

```diff
@@ -133,8 +133,8 @@ void ModeratorTzero::exec() {
   // iterate over the spectra
   for (int i = 0; i < static_cast<int>(numHists); ++i) {
     PARALLEL_START_INTERUPT_REGION
-    MantidVec &inbins = inputWS->dataX(i);
-    MantidVec &outbins = outputWS->dataX(i);
+    auto &inbins = inputWS->mutableX(i);
+    auto &outbins = outputWS->mutableX(i);
```

Comment:
I think we can do better here: Look at all the code blocks below, there are several places that set `outbins = inbins`, i.e., we can set up sharing. Probably just moving variables closer to where they are used will show the right way to do this.